File: 0169-majority-element/0169-majority-element.cs
public class Solution {
    public int MajorityElement(int[] nums) {
        int n = nums.Length, maxCnt = 0, majority = 0;
        var dict = new Dictionary<int,int>();
        for(int i = 0;i<n;i++){
            if(!dict.ContainsKey(nums[i])){
                dict.Add(nums[i],0);
            }
            dict[nums[i]]++;
        }
        foreach(var item in dict){
            if(item.Value > maxCnt){
                maxCnt = item.Value;
                majority = item.Key;
            }
        }
        return majority;
    }
}

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n = nums.size();
        int ele, cnt = 0;
        for(int i = 0;i<n;i++){
            if(cnt == 0){
                ele = nums[i];
                cnt = 1;
            }else if(nums[i] == ele) cnt++;
            else cnt--;
        }
        int cnt2 = 0;
        for(int i = 0;i<n;i++){
            if(ele == nums[i])cnt2++;
            if(cnt2 > (n/2)) return nums[i];
        }
        return -1;
    } 
};